scripts/lint-style.py

```diff
@@ -195,6 +196,30 @@ def four_spaces_in_second_line(lines, path):
         newlines.append((next_line_nr, new_next_line))
     return errors, newlines
 
+def nonterminal_simp_check(lines, path):
+    errors = []
+    newlines = []
+    annotated_lines = list(annotate_comments(lines))
+    for (line_nr, line, is_comment), (_, next_line, _) in zip(annotated_lines,
+                                                              annotated_lines[1:]):
+        # Check if the current line matches whitespace followed by "simp"
+        new_line = line
+        if (not is_comment) and re.search(r"^\s*simp$", line):
```

Comment:
I imagine that this is a regular expression that matches lines consisting of "<whitespace>simp<line_break>".

Shouldn't the linter also catch "<whitespace>simp [some lemmas, possibly with line breaks]" that are non-terminal?

Reply:
Good point.

Reply:
Seems tricky to match multiple lines like this, since the loop goes line by line, but I'll at least try to match simp [ ... ] \n

Reply:
Feel free to ignore this comment, but there are also a few non-terminal `simp ([...]) at ...`.  :speak_no_evil:

Reply:
Wait is it considered nonterminal to `simp at`? I thought it was ok because you can never finish a proof by `simp at` anyway.

Reply:
I would definitely think so: a change in the `simp` set could them change what the targeted hypothesis reduces to, breaking the rest of the proof.